### PR TITLE
Add Buttons

### PR DESCRIPTION
**Pull request by aesha-spreadd**
1) add 4 different types of buttons

1) list buttons
Screenshot 2023-12-21 at 10 01 58 PM

2) outlined buttons
Screenshot 2023-12-21 at 10 02 43 PM

3) text buttons
Screenshot 2023-12-21 at 10 03 08 PM

4) icon buttons
Screenshot 2023-12-21 at 10 03 38 PM

- [x]  Checking the code is duplicate or not?
- [x]  Are you going to merge this or not?
- []  Is code is very by shubhum sir or ankit sir?

### CODE REVIEW

File: README.md
Comment (shubham-spreadd) [inline README.md:1]: can you please explain this read me file

File: src/Button/Button.Styled.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  font-weight: 600;
+  text-transform: uppercase;
+
+  ${({ type, outline, border }) => css`
+    background-color: ${outline ? 'transparent' : theme?.[type]};
+    color: ${outline ? theme?.[type] : 'white'};
+    border-color: ${border ? 'transparent' : theme?.[type]};
+  `}
+
+  ${css`
```
Comment (shubham-spreadd) [inline src/Button/Button.Styled.js]:
```
/* Dynamic styles */
  ${({ type, outline, border, hovers, theme }) => {
    const bgColor = outline ? 'transparent' : theme?.[type];
    const textColor = outline ? theme?.[type] : 'white';
    const borderColor = border ? 'transparent' : theme?.[type];

    return css`
      background-color: ${bgColor};
      color: ${textColor};
      border-color: ${borderColor};

      /* Hover effect for outline buttons */
      ${outline &&
      hovers &&
      css`
        &:hover,
        &:active {
          background-color: ${theme?.[type]};
          color: white;
        }
      `}
    `;
  }}
```

can be manged like this!

File: src/Button/Button.stories.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      variant="outlined"
+      text="Secondary Button"
+      outline
+      hovers
+    />
+    <Button
+      type="success"
+      variant="outlined"
+      text="Success Button"
+      outline
```
Comment (shubham-spreadd) [inline src/Button/Button.stories.js]: outline should be a variant, not a separate prop.

File: src/Button/Button.stories.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  </div>
+);
+
+export const TextButtons = () => (
+  <div>
+    <Button
+      type="primary"
+      variant="contained"
+      text="Primary Link Button"
+      border
```
Comment (shubham-spreadd) [inline src/Button/Button.stories.js]: same here. border should be managed from varients only.

File: src/Button/Button.stories.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      outline
+    />
+  </div>
+);
+
+export const IconButtons = () => (
+  <div>
+    <Button
+      type="primary"
+      variant="outlined"
```
Comment (shubham-spreadd) [inline src/Button/Button.stories.js:41]: variant will be IconButton for the icon component.

**Review by shubham-spreadd — CHANGES_REQUESTED** [review]

File: src/Button/Button.stories.js (hunk 3 above)
Comment (aesha-spreadd) [reply src/Button/Button.stories.js]: Comment resolved

File: src/Button/Button.stories.js (hunk 4 above)
Comment (aesha-spreadd) [reply src/Button/Button.stories.js:41]: Comment resolved

File: src/Button/Button.stories.js (hunk 2 above)
Comment (aesha-spreadd) [reply src/Button/Button.stories.js]: comment resolved

File: src/Button/Button.Styled.js (hunk 1 above)
Comment (aesha-spreadd) [reply src/Button/Button.Styled.js]: comment resolved

File: src/Button/Button.Styled.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  ${({ type, variant, colorEffect, hovers }) => {
+    const isOutlined = variant === 'outlined';
+    const isTextButton = variant === 'textButton';
+    const bgColor = colorEffect ? 'transparent' : theme?.[type];
+    const textColor = colorEffect ? theme?.[type] : 'white';
+    const borderColor = isOutlined
+      ? theme?.[type]
+      : isTextButton
+      ? 'transparent'
+      : theme?.[type];
```
Comment (shubham-spreadd) [inline src/Button/Button.Styled.js]: avoid nested ternary

File: src/theme.js [hunk 6]
```diff
@@ -20,9 +20,9 @@ const theme = {
   inputBorder: 'rgb(74, 74, 104)',
   textColor: 'rgb(246, 249, 252)',
   successColor: 'rgb(76, 175, 80)',
-  dangerColor: 'rgb(244, 67, 54)',
+  danger: 'rgb(244, 67, 54)',
   secondaryColor: 'rgb(255, 152, 0)',
-  horizontalTitle: 'rgb(255, 78, 84);',
+  horizontalTitle: 'rgb(255, 78, 84)', // Remove the semicolon
```
Comment (shubham-spreadd) [inline src/theme.js]: I don't think this comment is required.

**Review by shubham-spreadd — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (shubham-spreadd) [inline .gitignore:1]: git ignore is not proper

**Review by shubham-spreadd — CHANGES_REQUESTED** [review]

File: src/Button/Button.stories.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import { FaSave } from "react-icons/fa";
+import Button from "./Button";
+
+export default {
+  title: "Button",
+  component: Button,
+};
+
+export const listButton = () => (
```
Comment (shubham-spreadd) [inline src/Button/Button.stories.js]: function names should be capital

**Review by shubham-spreadd — CHANGES_REQUESTED** [review]

**Review by shubham-spreadd — APPROVED** [review]

**Pull request merged**